src/config.rs

```diff
@@ -48,15 +48,127 @@ fn default_daemon() -> bool {
     false
 }
 
+/// RPC authentication options.
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq)]
+pub enum BitcoindRpcAuth {
+    /// Path to bitcoind's cookie file.
+    CookieFile(PathBuf),
+    /// "USER:PASSWORD" for authentication.
+    UserPass(String, String),
+}
+
 /// Everything we need to know for talking to bitcoind serenely
-#[derive(Debug, Clone, Deserialize, Serialize)]
+#[derive(Debug, Clone)]
 pub struct BitcoindConfig {
-    /// Path to bitcoind's cookie file, to authenticate the RPC connection
-    pub cookie_path: PathBuf,
+    /// Authentication credentials for bitcoind's RPC server.
+    pub rpc_auth: BitcoindRpcAuth,
     /// The IP:port bitcoind's RPC is listening on
     pub addr: SocketAddr,
 }
 
+impl Serialize for BitcoindConfig {
```

Comment:
Thanks for this, much simpler :relieved: I hadn't considered using the `#[serde(flatten)]` attribute.

Reply:
I've now changed it to use serde attributes. The deserializer is slightly different so that it returns an error if both "cookie_path" and "auth" are set in the config file.